src/xrpld/app/tx/detail/Payment.cpp

```diff
@@ -567,17 +587,17 @@ Payment::doApply()
         // We choose the base reserve as our bound because it is
         // a small number that seldom changes but is always sufficient
         // to get the account un-wedged.
-        if (dstAccountID != account_)
+
+        // Get the base reserve.
+        XRPAmount const dstReserve{view().fees().accountReserve(0)};
+
+        if (dstAmount > dstReserve ||
+            sleDst->getFieldAmount(sfBalance) > dstReserve)
         {
-            if (!view().exists(keylet::depositPreauth(dstAccountID, account_)))
-            {
-                // Get the base reserve.
-                XRPAmount const dstReserve{view().fees().accountReserve(0)};
-
-                if (dstAmount > dstReserve ||
-                    sleDst->getFieldAmount(sfBalance) > dstReserve)
-                    return tecNO_PERMISSION;
-            }
+            if (auto err =
+                    verifyDepositPreauth(ctx_, account_, dstAccountID, sleDst);
```

Comment:
Is there a performance difference with this order of if's vs switching them?

Reply:
Can't be anything noticeable as now more lightweight check before more heavier